### DISCUSSION TITLE
PushAPI - is that supported?

### DISCUSSION THREAD

**Discussion by matcake** [cypress-io/cypress#24145] [Questions and Help]
Hi, I'd like to prepare tests to verify if my WebPush service properly sends notifcations to the browser, ServiceWorker receives all of them correctly and then triggers NotificationsAPI.

At this moment I've got:
1. ServiceWorker registered
2.  Subscription saved to the backend (FCM key)

Browser run via Cypress doesn't receive any push notifications. The same configuration works great with desktop Chrome/Edge.

The question: is the PushAPI supported?

**Comment by matcake** [comment]
Any suggestions?

**Comment by filiphric** [comment]
Long shot, but maybe you could look into [this plugin](https://github.com/kamranayub/cypress-browser-permissions)? It enables you to take control of different browser permissions including push notifications, so maybe it might help?

The testing of whether the subscription being saved in the backend is probably `cy.task()` can help you with. Cypress only run things in browser, but with `cy.task()` you can run any node script you want and wait for it to complete and be validated. so if you create a script that will connect to backend and retreive the information you need, you can hook it up to Cypress and validate.
